benchmark/comprehensive_benchmark.dart

```diff
@@ -94,165 +94,156 @@ class _Runtime {
-        return _AddOperation(_randomElement());
+        return (type, _randomElement());
       case _OperationType.removeIdx:
         if (_set.isEmpty) {
-          return _AddOperation(_randomElement());
+          return noop;
         }
-        return _RemoveIdxOperation(r.nextInt(_set.length));
+        return (type, r.nextInt(_set.length));
       case _OperationType.removeElement:
         if (_set.isEmpty) {
-          return _AddOperation(_randomElement());
+          return noop;
         }
-        return _RemoveElementOperation(_set.elementAt(r.nextInt(_set.length)));
+        return (type, _set.elementAt(r.nextInt(_set.length)));
       case _OperationType.removeWhere:
-        return _RemoveWhereOperation(_randomElement());
+        return (type, _randomElement());
       case _OperationType.visit:
-        return _VisitOperation(_randomElement());
+        return (type, _randomElement());
       case _OperationType.iterateThenAdd:
-        return _IterateThenAddOperation(_randomElement());
+        return (type, _randomElement());
       case _OperationType.iterateThenRemove:
-        return _IterateThenRemoveOperation(_randomElement());
+        return (type, _randomElement());
     }
   }
 
   int _randomElement() => r.nextInt(_maxElement) + 1;
 }
 
 enum _OperationType {
+  noop(_noopOperation),
   // when queued, generates a random element; then adds using `add`
-  add,
+  add(_addOperation),
   // when queued, selects a random index; then removes using `removeAt`
-  removeIdx,
+  removeIdx(_removeIdxOperation),
   // when queued, selects a random element; then removes using `remove`
-  removeElement,
+  removeElement(_removeOperation),
   // when queued, generates a random factor; then removes all elements with
   // that factor using `removeWhere`
-  removeWhere,
+  removeWhere(_removeWhereOperation),
   // when queued, generates a random factor; then finds the elements matching
   // that factor, using normal for iteration
-  visit,
+  visit(_visitOperation),
   // when queued, generates two random factors; iterates over the set,
   //finds elements that match the first factor, then multiplies them by
   //the second factor, queue adding the results with the `add` operation
-  iterateThenAdd,
+  iterateThenAdd(_iterateThenAddOperation),
   // when queued, generates a random factor; iterates over the set, finding
   // elements that match the factor, then queue their removal with
   // the `removeElement` operation
-  iterateThenRemove,
-}
-
-abstract class _Operation {
-  final _OperationType type;
+  iterateThenRemove(_iterateThenRemoveOperation),
+  ;
 
-  const _Operation(this.type);
+  final List<_Operation> Function(_Operation, _Runtime, OrderedSet<int>)
+      execute;
 
-  List<_Operation> execute(_Runtime runtime, OrderedSet<int> set);
+  const _OperationType(this.execute);
 }
 
-class _AddOperation extends _Operation {
-  final int element;
+typedef _Operation = (_OperationType, int);
```

Comment:
no proper value classes so we can't do sealed inheritance

Reply:
Maybe worth making a sealed class from it?

Reply:
humm as I said I don't see how I could, since it has to be a tuple

Reply:
Why does it have to be a tuple?